### PR TITLE
Making frontend changes to output tags in the register-environment

### PR DESCRIPTION
**Pull request by pryenugu**
1. Making frontend changes to output tags in the register-environment in Deploy RG. 
2. Also making changes to not persist in task context in consumer(of output variables) tasks.

### CODE REVIEW

File: Tasks/DeployAzureResourceGroup/Utility.ps1 [hunk 1]
```diff
@@ -487,6 +488,26 @@ function Instantiate-Environment
     Write-Verbose "Added the environmnent $outputVariable to output variable" -Verbose
 }
 
+function Get-AzureResourcesTags
```
Comment (kmkumaran) [inline Tasks/DeployAzureResourceGroup/Utility.ps1]: Would you also share the changes to Pester?

File: Tasks/DeployAzureResourceGroup/Utility.ps1 [hunk 2]
```diff
@@ -487,6 +488,26 @@ function Instantiate-Environment
...
+
+    $tagsList = New-Object 'system.collections.generic.dictionary[[string],[system.collections.generic.dictionary[[string],[string]]]]' 
+
+    foreach($resource in $resources)
+    {
+        $resourceName = $resource.Name
+        if(-not ($resource.Tags -eq $null or $resource.Tags.Count -eq 0))
+        {
+            $resourceFqdn = ($azureVmDetails[$resourceName]).fqdn
+            $tagsList.Add($resourceFqdn, $resource.Tags)
```
Comment (anaggar) [inline Tasks/DeployAzureResourceGroup/Utility.ps1]: resource fqdn is not unique we should use resource name as a key to get tags corresponding to that resource

**Comment by pryenugu** [comment]
Need to bump minimum agent version when we check-in post the next deployment.

**Comment by anaggar** [comment]
looks good now, just verify them and add tests to catch issues where one can change key in the dictionary

File: Tasks/DeployAzureResourceGroup/Utility.ps1 (hunk 1 above)
Comment (pryenugu) [inline Tasks/DeployAzureResourceGroup/Utility.ps1]: Will not add now, will add when we check in in the next sprint.

**Comment by bryanmacfarlane** [comment]
accept or close

**Comment by kmkumaran** [comment]
Closing old PR.

**Pull request closed without merging**